domain/domain_test.go

```diff
@@ -0,0 +1,193 @@
+	assert.Equal(t, "create-error-code", apiErr.Errors[0].Code)
+}
+
+func TestDomainUpdate(t *testing.T) {
+	id := "dmn_456"
+	name := "clerk.dev"
+	clerk.SetBackend(clerk.NewBackend(&clerk.BackendConfig{
+		HTTPClient: &http.Client{
+			Transport: &mockRoundTripper{
+				T:      t,
+				in:     json.RawMessage(fmt.Sprintf(`{"name":"%s"}`, name)),
+				out:    json.RawMessage(fmt.Sprintf(`{"id":"%s","name":"%s"}`, id, name)),
+				path:   fmt.Sprintf("/v1/domains/%s", id),
+				method: http.MethodPatch,
+			},
+		},
+	}))
+
+	dmn, err := Update(context.Background(), id, &UpdateParams{
+		Name: clerk.String(name),
+	})
+	require.NoError(t, err)
+	assert.Equal(t, id, dmn.ID)
+	assert.Equal(t, name, dmn.Name)
+}
+
+func TestDomainUpdate_Error(t *testing.T) {
+	clerk.SetBackend(clerk.NewBackend(&clerk.BackendConfig{
+		HTTPClient: &http.Client{
+			Transport: &mockRoundTripper{
+				T:      t,
+				status: http.StatusBadRequest,
+				out: json.RawMessage(`{
+  "errors":[{
+		"code":"update-error-code"
+	}],
+	"clerk_trace_id":"update-trace-id"
+}`),
+			},
+		},
+	}))
+
+	_, err := Update(context.Background(), "dmn_123", &UpdateParams{})
+	require.Error(t, err)
+	apiErr, ok := err.(*clerk.APIErrorResponse)
+	require.True(t, ok)
+	assert.Equal(t, "update-trace-id", apiErr.TraceID)
+	require.Equal(t, 1, len(apiErr.Errors))
+	assert.Equal(t, "update-error-code", apiErr.Errors[0].Code)
+}
+
+func TestDomainDelete(t *testing.T) {
+	id := "dmn_789"
+	clerk.SetBackend(clerk.NewBackend(&clerk.BackendConfig{
+		HTTPClient: &http.Client{
+			Transport: &mockRoundTripper{
+				T:      t,
+				out:    json.RawMessage(fmt.Sprintf(`{"id":"%s","deleted":true}`, id)),
+				path:   fmt.Sprintf("/v1/domains/%s", id),
+				method: http.MethodDelete,
+			},
+		},
+	}))
+
+	dmn, err := Delete(context.Background(), id)
+	require.NoError(t, err)
+	assert.Equal(t, id, dmn.ID)
+	assert.True(t, dmn.Deleted)
+}
+
+func TestDomainList(t *testing.T) {
+	clerk.SetBackend(clerk.NewBackend(&clerk.BackendConfig{
+		HTTPClient: &http.Client{
+			Transport: &mockRoundTripper{
+				T: t,
+				out: json.RawMessage(`{
+	"data": [{"id":"dmn_123","name":"clerk.com"}],
+	"total_count": 1
+}`),
+				path: "/v1/domains",
+			},
+		},
+	}))
+
+	list, err := List(context.Background(), &ListParams{})
+	require.NoError(t, err)
+	assert.Equal(t, int64(1), list.TotalCount)
+	assert.Equal(t, 1, len(list.Domains))
+	assert.Equal(t, "dmn_123", list.Domains[0].ID)
+	assert.Equal(t, "clerk.com", list.Domains[0].Name)
+}
+
+type mockRoundTripper struct {
+	T *testing.T
+	// Response status.
+	status int
+	// Response body.
+	out json.RawMessage
+	// If set, we'll assert that the request body
+	// matches.
+	in json.RawMessage
+	// If set, we'll assert the request path matches.
+	path string
+	// If set, we'll assert that the request method matches.
+	method string
+}
+
+func (rt *mockRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	if rt.status == 0 {
```

Comment:
💡 Mostly a suggestion, it might be easier to use a constructor for initializations: `newMockRoundTripper`